pkg/networking/pod_eni_info_resolver.go

```diff
@@ -164,17 +165,28 @@ func (r *defaultPodENIInfoResolver) resolvePodsViaCascadedLookup(ctx context.Con
 			}
 		}
 	}
+	if len(podsOnSageMakerHyperPod) > 0 {
+		eniInfoByPodKeySageMakerHyperPod, err := r.resolveViaCascadedLookup(ctx, podsOnSageMakerHyperPod, false, true)
+		if err != nil {
+			return nil, err
+		}
+		if len(eniInfoByPodKeySageMakerHyperPod) > 0 {
+			for podKey, eniInfo := range eniInfoByPodKeySageMakerHyperPod {
+				eniInfoByPodKey[podKey] = eniInfo
+			}
+		}
+	}
 	return eniInfoByPodKey, nil
 }
 
-func (r *defaultPodENIInfoResolver) resolveViaCascadedLookup(ctx context.Context, pods []k8s.PodInfo, isFargateNode bool) (map[types.NamespacedName]ENIInfo, error) {
+func (r *defaultPodENIInfoResolver) resolveViaCascadedLookup(ctx context.Context, pods []k8s.PodInfo, isFargateNode bool, isSageMakerHyperPodNode bool) (map[types.NamespacedName]ENIInfo, error) {
 	eniInfoByPodKey := make(map[types.NamespacedName]ENIInfo)
 	resolveFuncs := []func(ctx context.Context, pods []k8s.PodInfo) (map[types.NamespacedName]ENIInfo, error){
 		r.resolveViaPodENIAnnotation,
 		r.resolveViaNodeENIs,
 		// TODO, add support for kubenet CNI plugin(kops) by resolve via routeTable.
 	}
-	if isFargateNode {
+	if isFargateNode || isSageMakerHyperPodNode {
```

Comment:
Cam we change this to check for non-ec2 pod instead of having or condition for Fargate and Sagemaker pod? 
This way we will only need to send one boolean value in this function and make decision based on that.

Reply:
yeah. makes sense. Addressed in latest commit. thanks!